src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java

```diff
@@ -32,19 +32,29 @@
  * {@code String s = Integer.toString(i = 2);}.
  * </p>
  * <p>
- * Rationale: With the exception of {@code for} iterators and assignment in {@code while} idiom,
+ * Rationale: With the exception of loop idioms,
  * all assignments should occur in their own top-level statement to increase readability.
  * With inner assignments like the one given above, it is difficult to see all places
  * where a variable is set.
  * </p>
  * <p>
- * Note: Check allows usage of the popular assignment in {@code while} idiom:
+ * Note: Check allows usage of the popular assignments in loops:
  * </p>
  * <pre>
  * String line;
- * while ((line = bufferedReader.readLine()) != null) {
+ * while ((line = bufferedReader.readLine()) != null) { // OK
  *   // process the line
  * }
+ *
+ * int flag;
+ * for (;(flag = 1) &lt;= 0;) { // OK
```

Comment:
I find it odd that the while and do while use `bufferedReader` for their examples but the for loop does something else which I question if it really is a "popular" assignment. You assign flag a constant right before the condition check, while the others receive the result from a method that can be variable.

Reply:
https://dxtkastb.github.io/innerassignment/diff/Hbase/xref/home/kaustubh/ck/contribution-master/checkstyle-tester/repositories/Hbase/hbase-common/src/main/java/org/apache/hadoop/hbase/util/Base64.java.html#L950
I see this as a good example.

Reply:
> I find it odd that the while and do while use `bufferedReader` for their examples but the for loop does something else which I question if it really is a "popular" assignment. You assign flag a constant right before the condition check, while the others receive the result from a method that can be variable.

I thought using bufferedReader in examples is going a bit monotonous, nonetheless I will update the example

Reply:
the same code is used